Prog/LgUnit - Ipums Table 2B.sas

```diff
@@ -0,0 +1,543 @@
+	else if 1400 <= &var1. < 1500 then &var2. = 14;
+	else if 1500 <= &var1. < 1600 then &var2. = 15;
+%mend recode_dvars;
+
+
+%macro ipums_lgunit (tenure,yrs);
+
+%let ten = %upcase( &tenure. );
+%let yrs = %upcase( &yrs. );
+
+%if &yrs. = ACS %then %do;
+	%let indata = Acs_2012_16_dc;
+	%let vacdata = acs_2012_16_vacant_dc;
+%end;
+
+%if &yrs. = 2000 %then %do;
+	%let indata = Ipums_2000_dc;
+	%let vacdata = Ipums_2000_vacant_dc;
+%end;
+
+data vacdata;
+	set ipums.&vacdata. ;
+
+	%if &ten. = OWN %then %do;
+		if vacancy in (2,4);
+	%end;
+
+	%else %if &ten. = RENT %then %do;
+		if vacancy in (1,3);
+	%end;
+run;
+
+data hhwts;
+	set ipums.&indata. (where=(pernum=1 and gq in (1,2)))  
+		vacdata ;
+	keep serial puma hhwt;
+
+	%if &yrs. = 2000 %then %do;
+
+	if ownershd in (12,13) then ownershp = 1;
+	else if ownershd in (21,22) then ownershp = 2;
+
+	%end;
+
+
+	%if &ten. = OWN %then %do;
+		if ownershp = 1;
+	%end;
+
+	%else %if &ten. = RENT %then %do;
+		if ownershp = 2;
+	%end;
+run;
+
+proc sort data = hhwts; by serial; run;
+
+
+
+data pretables;
+	set Ipums.&indata.;
+
+	%if &yrs. = 2000 %then %do;
+		
+	%recode_dvars (raced,race);
+	%recode_dvars (hispand,hispan);
+	%recode_dvars (related,relate);
+
+	if ownershd in (12,13) then ownershp = 1;
+	else if ownershd in (21,22) then ownershp = 2;
+
+	/* Recode builtyr to match builtyr2 from ACS data */
+
+	if builtyr in (1,2,3,4) then builtyr2 = 6;
+		else if builtyr in (5,6) then builtyr2 = 4;
+		else if builtyr in (7,8) then builtyr2 = 2;
+		else if builtyr in (9) then builtyr2 = 1;
+
+	%end;
+
+
+	%if &ten. = OWN %then %do;
+		if ownershp = 1;
+	%end;
+
+	%else %if &ten. = RENT %then %do;
+		if ownershp = 2;
+	%end;
+
+	 /*Keep only HHs*/
+	if gq in (1,2);
+
+	/* All households */
+	allhh = 1;
+
+	/* Householder age */
+	if pernum = 1 then hher_age = age;
+
+	/* Gousehold income */
+	if pernum = 1 then hh_inc = hhincome;
+
+    /* Flag large units*/
+	if numprec >= 4 then largeunit = 1;
+		else largeunit = 0;
+
+	/* Adult*/
+	if age>=18 then isadult = 1;
+	    else isadult = 0;
+
+	numadults = isadult;
+
+	/*adult but not senior*/
+	if 18<=age<65 then nonsenioradult = 1;
+        else nonsenioradult = 0;
+   
+	/*seniors*/
+    if age>=65 then issenior = 1;
+	    else issenior = 0;
+
+    /*schoolage children*/
+    if 6<age<18 then isschoolage = 1;
+	    else isschoolage = 0;
+
+	/*all kids */
+	if age <18 then iskid = 1;
+		else iskid =0;
+
+	if iskid = 1 then kid_age = age;
+
+	numkids = iskid;
+
+	/*Disability*/
+	isdis= 0;
+    if diffmob = 2 or diffcare = 2 then isdis = 1;
+
+	array vars {*} diffeye diffhear diffphys diffrem;
+	array f_vars {*} f_diffeye f_diffhear f_diffphys f_diffrem;
+
+	do i = 1 to 4;
+           if vars{i} = 2 then f_vars{i} = 1;
+           else f_vars{i} = 0;
+    end;
+
+    if sum(f_diffhear, f_diffeye, f_diffphys, F_diffrem)>=2 then isdis = 1;
+
+	/*3 generation household*/
+
+	/* Child or stepchild */
+	if relate in (3,4) then before1gen = 1;
```

Comment:
include foster children?

Reply:
Added foster children.

Reply:
@rpitingolo  = actually i was just thinking - can you code something that's like other relative and age < 18 ?  (just thinking that many of our tax clients have nieces, nephews, cousins, etc. that they are taking care of).

Reply:
OK, added that in.